src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -90,6 +90,24 @@ class RsErrorAnnotator : Annotator, HighlightRangeExtension {
         checkReferenceIsPublic(field, o, holder)
     }
 
+    private fun checkConstantUsageIsSafe(ref: RsReferenceElement, holder: AnnotationHolder) {
+        val element = ref.reference.resolve() as? RsConstant ?: return
+        val expr = ref.ancestorOrSelf<RsExpr>() ?: return
+
+        val isInUnsafeCx = expr.isInUnsafeBlockOrFn()
+        if (isInUnsafeCx) {
+            return
+        }
```

Comment:
Looks like `if (expr.isInUnsafeBlockOrFn()) return` is not less readable but shorter